tests/pytest/test_navs.py

```diff
@@ -1,22 +1,23 @@
 """Tests for """
 
 
+import contextlib
 import random
 import textwrap
-from typing import Any, Callable
+from typing import Generator
 
 from htmltools import TagList
 
 from shiny import ui
 from shiny._utils import private_seed
-from shiny.ui._navs import NavSet
 
 
 # Fix the randomness of these functions to make the tests deterministic
-def with_private_seed(func: Callable[[], NavSet], *args: Any, **kwargs: Any):
+@contextlib.contextmanager
+def private_seed_n(n: int = 0) -> Generator[None, None, None]:
     with private_seed():
-        random.seed(0)
-        return func(*args, **kwargs)
+        random.seed(n)
+        yield
```

Comment:
The purpose of this new context manager is to make it so the code inside the `with private_seed_n():` statement is clearer -- it will have the same form as it would if were called without the `with private_seed_n():`.